### ISSUE TITLE
Add options --storage-location and --from-storage-location to the velero restore command

### ISSUE THREAD

**Issue by konvergence** [vmware-tanzu/velero#3026]
**Describe the problem/challenge you have**

  1.  Velero allow to add backup-location in ReadOnly to be able to restore from antoher cluster. But if you have already a backupName on the "default" backup-location you can not see the other backupName on the alternative backup-location. (This for me a bug)
  2. for a `velero restore` we should be able to choose the backupName `--from-storage-location` and be able to choose on which `--storage-location` we want to write the  restic `/restore` information (see #3019)
 

**Describe the solution you'd like**
  1. Add options `--storage-location` and `--from-storage-location` to the velero restore command
  2.  Update the sync controller to allow to have same backupName but on different backup-location

**Environment:**

- Velero version (use `velero version`): 

```
Client:
        Version: v1.5.1
        Git commit: 87d86a45a6ca66c6c942c7c7f08352e26809426c
Server:
        Version: v1.5.1
```
- Velero features (use `velero client config get features`): 

```
features: <NOT SET>
```

- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes installer & version:` rancher 2.4.5 rke

- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

```
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
````
**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by zubron** [comment]
Hi @konvergence,

> Velero allow to add backup-location in ReadOnly to be able to restore from antoher cluster. But if you have already a backupName on the "default" backup-location you can not see the other backupName on the alternative backup-location. (This for me a bug)

Just to clarify, do you have backups with the same name in both buckets, but you are only able to see the backup from the default bucket?

**Comment by konvergence** [comment]
Thanks 
Yes a had the same backupName on the 2 locations. 

Is there a way to configure a prefix for backupName on velero server side ?

**Comment by zubron** [comment]
Hi again @konvergence. Unfortunately the backups are considered unique based on their name, not a combination of the name and the storage location. When you add a new backup storage location where a backup name matches a backup in the existing storage location, it will not be imported into the new cluster and currently we don't have a way to work around that.

Given that this is a distinct issue from point 2 in the description above, I am going to open a separate issue to address this and leave this issue to solely address the redirecting where the restore information is written to.

**Comment by zzmg** [comment]
hi
i face the same problem with you
I cannot migrate from one cluster to another cluster right now，because new cluster do not get  backup object from a new backup location，and old cluster' s backup in the new backup location

**Comment by eleanor-millman** [comment]
Closing because will hopefully be addressed around 2.0 or a bit later by other changes (2022 or later).

**Comment by jsalatiel** [comment]
Any progress on this after 2 years? Still not possible to migrate when the user on the source bucket has only read permissions.
